ocean_lib/ocean/test/test_ocean_assets.py

```diff
@@ -38,43 +42,166 @@ def test_register_asset(publisher_ocean_instance, publisher_wallet, consumer_wal
     assert ocn.assets.resolve(invalid_did) is None
 
 
-@pytest.mark.skip(reason="TODO: update function on OceanAssets class")
-def test_update(publisher_ocean_instance, publisher_wallet, consumer_wallet, config):
-    ocn = publisher_ocean_instance
-    block = ocn.web3.eth.block_number
-    alice = publisher_wallet
-    bob = consumer_wallet
+def test_update(publisher_ocean_instance, publisher_wallet, config):
+    """Helper function for asset update based on ddo_sa_sample.json."""
+    ocean = publisher_ocean_instance
+
+    _, erc20_token = deploy_erc721_erc20(
+        publisher_ocean_instance.web3, config, publisher_wallet, publisher_wallet
+    )
+
+    ocean = publisher_ocean_instance
 
-    ddo = create_asset(ocn, alice, config)
+    data_provider = DataServiceProvider
+    file1_dict = {"type": "url", "url": "https://url.com/file1.csv", "method": "GET"}
+    file1 = FilesTypeFactory(file1_dict)
+    encrypt_response = data_provider.encrypt(
+        [file1], "http://172.15.0.4:8030/api/services/encrypt"
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
 
-    # Publish the metadata
-    _name = "updated name"
-    ddo.metadata["name"] = _name
-    assert ddo.metadata["name"] == _name, "Asset's name was not updated."
+    ddo = create_asset(publisher_ocean_instance, publisher_wallet, config)
 
-    with pytest.raises(ValueError):
-        ocn.assets.update(ddo, bob)
+    new_metadata = copy.deepcopy(ddo.metadata)
 
-    _ = ocn.assets.update(ddo, alice)
-    block_confirmations = ocn.config.block_confirmations.value
+    # Update only metadata
+    _description = "Updated description"
+    new_metadata["description"] = _description
+    new_metadata["updated"] = datetime.now().isoformat()
+    ddo.metadata = new_metadata
 
-    log = ddo_reg.get_event_log(
-        ddo_reg.EVENT_METADATA_UPDATED, block - block_confirmations, asset_id, 30
+    ocean.assets.update(
+        asset=ddo,
+        publisher_wallet=publisher_wallet,
     )
-    assert log, "no ddo updated event"
-    _asset = wait_for_update(ocn, ddo.did, "name", _name)
+
+    _asset = wait_for_update(ocean, ddo.did, "description", _description)
     assert _asset, "Cannot read asset after update."
-    assert (
-        _asset.metadata["main"]["name"] == _name
-    ), "updated asset does not have the new updated name !!!"
+    assert _asset.datatokens == ddo.datatokens
+    assert len(_asset.services) == len(ddo.services)
+    assert _asset.services[0].as_dictionary() == ddo.services[0].as_dictionary()
+    assert _asset.credentials == ddo.credentials
+
+    # Update credentials
+    new_asset = new_metadata = copy.deepcopy(_asset)
+    new_metadata = copy.deepcopy(_asset.metadata)
+    _description = "Updated credentials"
+    _new_credentials = {
+        "allow": [{"type": "address", "values": ["0x123", "0x456"]}],
+        "deny": [{"type": "address", "values": ["0x2222", "0x333"]}],
+    }
+    new_metadata["description"] = _description
```

Comment:
let's separate the tests a bit. If we are testing for metadata update above, please only check credentials update now. If we always test metadata + something, we might get false results. I want to see that editing the credentials without any metadata updates works properly. Same for the other cases.